content/server/soju.md

```diff
@@ -0,0 +1,194 @@
+chgrp soju /etc/letsencrypt/live/{{<hl>}}irc.example.org{{</hl>}}/privkey.pem
+```
+
+Now let's automate soju's restart each time we pull a new cert:
+
+Write out this script to /etc/letsencrypt/renewal-hooks/post/soju.sh:
+
+```sh
+#!/bin/sh -eu
+systemctl reload soju
+```
+
+Then make it executable:
+
+```sh
+chmod +x /etc/letsencrypt/renewal-hooks/post/soju.sh
+```
+
+## Configuration
+
+### Soju
+
+Now let's fill out a configuration file at /etc/soju/config:
+
+```scfg
+listen        ircs://
+# optional: add unix socket to control soju through the cli
+listen        unix+admin://
+
+tls           /etc/letsencrypt/live/{{<hl>}}irc.example.org{{</hl>}}/fullchain.pem /etc/letsencrypt/live/{{<hl>}}irc.example.org{{</hl>}}/privkey.pem
+db            sqlite3 /var/lib/soju/main.db
+message-store db
+hostname      {{<hl>}}irc.example.org{{</hl>}}
+auth          internal
+title         "anon's soju bouncer"
+```
+
+Make an admin user called anon:
+
+```sh
+sojudb create-user anon -admin
+```
+
+### Systemd Service
+
+Then let's make a service in /etc/systemd/system/soju.service
+
+```systemd
+[Unit]
+Description=soju IRC bouncer service
+Documentation=https://soju.im/
+Documentation=man:soju(1) man:sojuctl(1)
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+Type=simple
+User=soju
+Group=soju
+DynamicUser=yes
+StateDirectory=soju
+ConfigurationDirectory=soju
+RuntimeDirectory=soju
+AmbientCapabilities=CAP_NET_BIND_SERVICE
+ExecStart=/usr/local/bin/soju
+ExecReload=/bin/kill -HUP $MAINPID
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+```
+
+Then enable our new service
+
+```sh
+systemctl daemon-reload
+systemctl enable --now soju
+```
+
+## Usage
+
+### Client Configurations
+
+Soju can be used by most legacy irc clients, but ones that support the soju.im/bouncer-networks extension make multiple servers work seamlessly.
+
+Here is a table of recommended clients that support said extension.
+
+| Client                                                      | Interface    |
+| :---------------------------------------------------------- | :----------  |
+| [Senpai](https://sr.ht/~delthas/senpai/)                    | Terminal/TUI |
+| [Goguma](https://f-droid.org/packages/fr.emersion.goguma/)  | Android/iOS  |
+| [Gamja](https://sr.ht/~emersion/gamja/)                     | Web          |
+| [Chathistorysync](https://sr.ht/~emersion/chathistorysync/) | Plain Text   |
+
+See more client configurations [here](https://git.sr.ht/~emersion/soju/tree/master/item/contrib/clients.md) for legacy clients.
+
+Here is an example client configuration for senpai.
+
+```scfg
+address  {{<hl>}}irc.example.org{{</hl>}}
+nickname anon
+password <password for anon>
+```
+
+### New Networks
+
+Several of the clients above have special user interfaces to add new servers, but the default way to do it with any client is with the `network create` command.
+
+For example, this is how we'd add [libera.chat](https://libera.chat) to our server:
```

Comment:
```suggestion
Several of the clients above have special user interfaces to add new servers, but the default way to do it with any client is with the `network create` command.
You can control soju with these commands, which you send to the soju BouncerServ user with `/msg`.

For example, this is how we'd add [irc.libera.chat](https://libera.chat) to our server:
```

Reply:
the `irc.` subdomain is the only one of your corrections I disagree with. The link is to libera's `https://` site, not `ircs://`.

Reply:
I just put that there because libera.chat IRC network is served over the irc.libera.chat domain, it's alright if you disagree